graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/sdl/parsing/SDLTypeChecker.java

```diff
@@ -153,9 +155,27 @@ private static SDLDefinitionStatus checkForFieldsConsistency(ObjectTypeDefinitio
                     l.add(fieldDefinition.getType().toString());
                 }
             });
+            //Verify that type exists in definition registry
             if (!l.isEmpty()) {
                 return new SDLDefinitionStatus(objectTypeDefinition.getName(), SDLDefinitionStatusType.MISSING_TYPE, StringUtils.join(l, ','));
             }
+            //Validate that if custom fetchers are declared (through fetcher directive) they exist in the map of registered fetchers.
+            SDLSchemaService service = BundleUtils.getOsgiService(SDLSchemaService.class, null);
```

Comment:
Would it not be better to pass the SDLSchemaService as a parameter instead of looking up viaBundleUtils ?

Reply:
Good point, I have just updated the code to reflect your suggestion.